### PR TITLE
configParser1.0

### PR DESCRIPTION
**Pull request by maartenRamadan**
#1

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -1,12 +1,17 @@
 NAME=not-apache
-INCLUDES=./includes
-FLAGS=-Wall -Wextra -Werror -std=c++98 -pedantic-errors
+INCLUDES=./includes/*.hpp
+FLAGS=-Wall -Wextra -Werror -std=c++98 -pedantic-errors -g -fsanitize=address
 CC=clang++
 
-HEADERS=
+HEADERS=*.hpp
```
Comment (mrjvs) [inline makefile]: Change INCLUDES variable to `./src/includes` and explicitly name all headers in HEADERS (just like in FILES)

File: makefile [hunk 2]
```diff
@@ -16,6 +21,7 @@ all: $(NAME)
 
 $(NAME): $(OBJS) $(HEADERS)
 	$(CC) -I $(INCLUDES) $(FLAGS) -o $(NAME) $(OBJS)
+	rm -f $(OBJS)
```
Comment (mrjvs) [inline makefile]: dont

File: makefile (hunk 1 above)
Comment (mrjvs) [reply makefile]: Oh and move all headers to ./src/includes as well

File: makefile [hunk 3]
```diff
@@ -1,12 +1,17 @@
 NAME=not-apache
-INCLUDES=./includes
-FLAGS=-Wall -Wextra -Werror -std=c++98 -pedantic-errors
+INCLUDES=./includes/*.hpp
+FLAGS=-Wall -Wextra -Werror -std=c++98 -pedantic-errors -g -fsanitize=address
```
Comment (pmerku) [inline makefile]: Debug flags should be put in another ruleset

File: src/Server.cpp [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+Server::Server(int fd) {
+	(void)fd;
+}
+
+Server::~Server() {}
+
+Server::Server(const Server& rhs)
+{
+	(void)rhs;
+}
```
Comment (mrjvs) [inline src/Server.cpp]: Whats the point of all of these? they do nothing?

File: src/Location.cpp [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+Location::Location(const Location& rhs) {
+	(void)rhs;
+}
+
+Location::~Location() {}
+
+Location&	Location::operator=(const Location& rhs) {
+	(void)rhs;
+	return *this;
+}
```
Comment (pmerku) [inline src/Location.cpp]: If not needed not neccessary to implement (we don't have to do coplien)

File: src/ConfigParser.cpp [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+	while (ft::getLine(fd, buf))
+	{
+		if (buf == "server {")
+		{
+			Server* tmp = new Server;
+			tmp->parse(fd);
+			servers.push_back(tmp);
+		}
+	}
+}
```
Comment (mrjvs) [inline src/ConfigParser.cpp]: Please use a namespace for all config parsing (except for ft::)
and move the headers into a seperate folder in includes.
and move the source files into a seperate folder in the src.

**Review by mrjvs — CHANGES_REQUESTED** [review]

File: src/includes/ConfigParser.hpp [hunk 7]
```diff
@@ -0,0 +1,20 @@
+#pragma once
```
Comment (pmerku) [inline src/includes/ConfigParser.hpp:1]: To be inline with the rest of the code, change this to a normal header guard

**Review by pmerku — CHANGES_REQUESTED** [review]

**Pull request closed without merging**